chore: Documentation on how the CLI works (#448)
subconsciousnetwork/noosphere: rust/noosphere-cli/README.md

The Noosphere CLI is a reference client and pedagogical tool to demonstrate
the principles of the Noosphere protocol and give interested users a
no-code, low-complexity tool to synchronize content through the Noosphere.

## Usage
### User perspective
```sh
# Create an identity
orb key create `whoami`

# make a directory for your sphere.
mkdir my-sphere
cd my-sphere

# Generate your personal sphere.
orb sphere create --owner-key `whoami`

# ..now make edits ..

# see the status of files in your directory
orb status

# persist changes to the sphere
orb save

> # join a gateway, after you've set this identity as the counterpart

Reviewer: "join" means something else in this context e.g. in `orb sphere join`

Author: How do you mean? From the linked video, this is what was typed to join the gateway.

Reviewer: i just mean the comment "join a gateway" in a pedantic-sense that maybe there's a better verb here (e.g. "register|subscribe to|set a gateway"), as we use `orb sphere join` later where "join" means synchronize another sphere (which is kinda what's going on here, admittedly)

> # join a gateway, after you've set this identity as the counterpart

Reviewer: Wonder if we can just get the counterpart sphere from the gateway after setting, though makes things a bit magical. In lieu of that, maybe just have a line for that like `orb config set counterpart <counterpart_did>`

Reviewer 2: Yah, this is room for refinement. Today it doesn't work that way, so the docs are accurate.

orb config set gateway-url <gatewayurl>

# sync your changes with the upstream gateway
orb sync

# sync data from a different sphere. Note, you'll need to
# `orb auth add <did>` from that other sphere.

> orb sphere join <their DID> --local-key `whoami`

Reviewer 2: Sorry, just noticed this but: I think you need to join first, and then sync (currently steps as documented are in reverse order).

Author: ```suggestion
orb sphere join <their DID> --local-key `whoami`
orb sync # pull in their changes
```
It was intended as a different sync.

The Noosphere CLI is a reference client and pedagogical tool to demonstrate
the principles of the Noosphere protocol and give interested users a
no-code, low-complexity tool to synchronize content through the Noosphere.

## Usage
### User perspective
```sh
# Create an identity
orb key create `whoami`

# make a directory for your sphere.
mkdir my-sphere
cd my-sphere

# Generate your personal sphere.
orb sphere create --owner-key `whoami`

# ..now make edits ..

# see the status of files in your directory
orb status

# persist changes to the sphere
orb save

> # join a gateway, after you've set this identity as the counterpart

Author: ```suggestion
# link your sphere to a gateway by setting this identity as the counterpart
```
